<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DeveloperWidget</class>
 <widget class="QWidget" name="DeveloperWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1472</width>
    <height>956</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidgetDevelopment">
     <property name="currentIndex">
      <number>11</number>
     </property>
     <property name="documentMode">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="tabComms">
      <attribute name="title">
       <string>Comms</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="1">
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>432</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="0">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>662</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButtonStartConstilation">
         <property name="text">
          <string>Start Constalation</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>661</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="1">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>432</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabTriggers">
      <attribute name="title">
       <string>Triggers</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="TriggerManagerWidget" name="widgetTriggerManager" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <spacer name="horizontalSpacer_5">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabGait">
      <attribute name="title">
       <string>Gait</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_10">
       <item>
        <widget class="GaitWidget" name="widgetGait" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabNetwork">
      <attribute name="title">
       <string>Network</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QGroupBox" name="groupBoxConnect">
         <property name="title">
          <string>Connect</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="pushButtonTest">
              <property name="text">
               <string>TEST</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="LocalAddressEntry" name="comboBoxLocalAddress">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="TextEntry" name="lineEditBindPort">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="toolTip">
               <string>Select on which port this hub should listen for incomming connections by remotes and agents</string>
              </property>
              <property name="maxLength">
               <number>6</number>
              </property>
              <property name="placeholderText">
               <string>Listen port</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="TryToggle" name="tryToggleListen" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>10</width>
                <height>10</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Click to toggle server listening</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>112</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabHUD">
      <attribute name="title">
       <string>HUD</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="HUDWidget" name="widgetHUD" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="IdentityAssetEditor" name="tabIdentity">
      <attribute name="title">
       <string>Identity</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_8"/>
     </widget>
     <widget class="QWidget" name="tabSerial">
      <attribute name="title">
       <string>Serial</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="SerialSettingsWidget" name="widgetSerialSettings" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabCamera">
      <attribute name="title">
       <string>Camera</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="CameraWidget" name="widgetCamera" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabZoo">
      <attribute name="title">
       <string>Zoo</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_12">
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>102</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabCL">
      <attribute name="title">
       <string>OpenCL</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButtonCLGLRender">
         <property name="toolTip">
          <string>Start the currently selected OpenCL rendering engine to produce images.</string>
         </property>
         <property name="text">
          <string>Render</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButtonCLGLDisplay">
         <property name="toolTip">
          <string>Start transferring data from OpenCL via OpenCL-OpenGL interop to display in OpenGL context</string>
         </property>
         <property name="text">
          <string>Display</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="CLGLView" name="openGLWidgetCLGLView"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabCreepyVoice">
      <attribute name="title">
       <string>CreepyVoice</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_11">
       <item>
        <widget class="SpeechControlWidget" name="widgetSpeechControl" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabActuators">
      <attribute name="title">
       <string>Actuators</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="2" column="0">
        <widget class="QSplitter" name="splitter_3">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <widget class="ActuatorManagementWidget" name="widgetActuatorControl" native="true"/>
         <widget class="PoseMappingWidget" name="widgetPoseMapping" native="true"/>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabTest">
      <attribute name="title">
       <string>Test</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_13">
       <item>
        <widget class="QListWidget" name="listWidgetRichTest"/>
       </item>
      </layout>
     </widget>
     <widget class="WidgetIllustrationEditor" name="tabWidgetIllustration">
      <attribute name="title">
       <string>Widget Illustration</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_10"/>
     </widget>
     <widget class="InternalResourceManager" name="tabInternalResources">
      <attribute name="title">
       <string>Internal Resources</string>
      </attribute>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ActuatorManagementWidget</class>
   <extends>QWidget</extends>
   <header location="global">hardware/actuator/ActuatorManagementWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TryToggle</class>
   <extends>QWidget</extends>
   <header location="global">components/TryToggle.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SpeechControlWidget</class>
   <extends>QWidget</extends>
   <header location="global">voice/SpeechControlWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SerialSettingsWidget</class>
   <extends>QWidget</extends>
   <header location="global">hardware/serial/SerialSettingsWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GaitWidget</class>
   <extends>QWidget</extends>
   <header location="global">gait/GaitWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PoseMappingWidget</class>
   <extends>QWidget</extends>
   <header location="global">pose/PoseMappingWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CameraWidget</class>
   <extends>QWidget</extends>
   <header location="global">camera/CameraWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TextEntry</class>
   <extends>QLineEdit</extends>
   <header location="global">components/TextEntry.hpp</header>
  </customwidget>
  <customwidget>
   <class>LocalAddressEntry</class>
   <extends>QComboBox</extends>
   <header location="global">node/LocalAddressEntry.hpp</header>
  </customwidget>
  <customwidget>
   <class>HUDWidget</class>
   <extends>QWidget</extends>
   <header location="global">hud/HUDWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CLGLView</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">clt/CLGLView.hpp</header>
  </customwidget>
  <customwidget>
   <class>TriggerManagerWidget</class>
   <extends>QWidget</extends>
   <header location="global">trigger/TriggerManagerWidget.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>InternalResourceManager</class>
   <extends>QWidget</extends>
   <header location="global">manager/InternalResourceManager.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>IdentityAssetEditor</class>
   <extends>QWidget</extends>
   <header location="global">identity/IdentityAssetEditor.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>WidgetIllustrationEditor</class>
   <extends>QWidget</extends>
   <header location="global">illustration/WidgetIllustrationEditor.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonStartConstilation</sender>
   <signal>clicked()</signal>
   <receiver>DeveloperWidget</receiver>
   <slot>startConstilation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>736</x>
     <y>491</y>
    </hint>
    <hint type="destinationlabel">
     <x>735</x>
     <y>477</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>startConstilation()</slot>
 </slots>
</ui>
